utils/localStorage.js

```diff
@@ -0,0 +1,96 @@
+var MAX_LOCAL_QUERIES = 5;
+
+function saveSuggestionQueries(storageName, queries) {
+    queries = JSON.stringify(queries).replace(/<\/?b>/g, '');
+    window.localStorage.setItem(storageName, queries);
+}
+
+function getSuggestionQueries(storageName) {
+    try {
+        return JSON.parse(window.localStorage.getItem(storageName));
+    } catch (ignored) {
+        return [];
+    }
+}
+
+function removeQueryFromLocalStorage(storageName, term) {
+    delete term.isQueryHistory;
+    var queries = getSuggestionQueries(storageName);
+    var filteredQueries = queries.filter(function (query) {
+        return JSON.stringify(query) !== JSON.stringify(term).replace(/<\/?b>/g, '');
+    });
+    saveSuggestionQueries(storageName, filteredQueries);
+}
+
+function addQueryToLocalStorage(storageName, query) {
+    delete query.isQueryHistory;
+    var queries = getSuggestionQueries(storageName);
+    if (queries === null) {
+        saveSuggestionQueries(storageName, [query]);
+    } else {
+        queries = queries.filter(function(element){
+            return JSON.stringify(element) !== JSON.stringify(query).replace(/<\/?b>/g, '');
+        })
+        if(queries.length >= MAX_LOCAL_QUERIES){
+            queries.shift();
+        }
+        queries.push(query);
+        saveSuggestionQueries(storageName, queries);
+    }
+}
+
+function getQueriesFromLocalStorage(storageName, term) {
+    var queries = getSuggestionQueries(storageName);
+    if (queries !== null) {
+        var matchedQueries = queries.map(function (query) {
+            query = JSON.stringify(query);
+            var regex = new RegExp(`${term}`);
+            var match = regex.exec(query);
+            if (match) {
+                query = query.replace(match, `<b>${match}</b>`);
+                query = JSON.parse(query);
+                query.isQueryHistory = true;
+                return query;
+            }
+            return null;
+        });
+        matchedQueries = matchedQueries.filter(function (value) { return value !== null });
+        return matchedQueries;
+    }
+    return [];
+}
+
+
+function removeDuplicatedQueries(queries) {
+    var titles = [];
+    var cleanQueries = [];
+    for (var query of queries) {
+        var isLocal = query.isQueryHistory;
+        delete query.isQueryHistory;
+        var text = JSON.stringify(query).replace(/<\/?b>/g, '')
+        if (!titles.includes(text)) {
```

Comment:
The `Array.prototype.includes` is not supported by any IE, use [indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) instead
https://caniuse.com/#feat=array-includes
```suggestion
        if (titles.indexOf(text) === -1) {
```